src/test/ui/sanitize/memory.rs

```diff
@@ -15,15 +15,15 @@
 #![feature(start)]
 #![feature(test)]
 
-use std::hint::black_box;
+use std::hint::pretend_used;
 use std::mem::MaybeUninit;
 
 #[inline(never)]
 #[no_mangle]
 fn random() -> [isize; 32] {
     let r = unsafe { MaybeUninit::uninit().assume_init() };
     // Avoid optimizing everything out.
-    black_box(r)
+    pretend_used(r)
```

Comment:
`pretend_use` / blackbox / ... is the completely wrong function to call here right? It doesn't help with correctness and this whole code relies on the black box hiding the fact that it's UB to read uninitialized memory from the compiler. (Same in the address sanitizer test)

Reply:
So, I think what's going on here is actually that we _want_ the code to be incorrect, but without `pretend_used` (as the comment implies) the `r` would be entirely optimized out by the compiler. `pretend_used` here causes the compiler to leave `r` in, which in turn means that the undefined behavior can be detected by the sanitizer.